cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/UnsupportedRelationException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.relational;
+
+/**
+ * Used to denote when a Relation is not supported by an {@link ExpressionFactory} when trying to operate on columns.
+ */
+public class UnsupportedRelationException extends Exception {
```

Comment:
Please don't introduce much more new classes that extend directly from Exception. Use a common base class. Also I'd prefer to extend from Runtime. All current functional logic relies on unchecked exceptions and does not play well with checked ones